### ISSUE TITLE
Object Detection When Using Image Size 1280 in YOLOv8x Training

### ISSUE THREAD

**Issue by uzunkadir** [ultralytics/ultralytics#18849]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I am training the YOLOv8x model using the VisDrone dataset. I set the image size to 1280 because I want the model to detect both large and small objects.

My training code is as follows:
```
from ultralytics import YOLO

model = YOLO('yolov8x.pt')

model.train(
    data='data/VisDrone.yaml',
    epochs=50,
    imgsz=1280,
    batch=4,
    device=0,
    workers=8,
)
```

By using an image size of 1280 instead of the default 640, the model was able to detect smaller objects, and the mAP values increased as desired. The results are as follows:

```
Class                Images  Instances  Box(P)  R     mAP50  mAP50-95:
all                  548      38759      0.677   0.579  0.616   0.398
pedestrian           520      8844       0.76    0.644  0.719   0.385
people               482      5125       0.711   0.538  0.594   0.279
bicycle              364      1287       0.54    0.41   0.435   0.231
car                  515      14064      0.857   0.868  0.906   0.69
van                  421      1975       0.635   0.598  0.627   0.472
truck                266      750        0.659   0.552  0.573   0.415
tricycle             337      1045       0.6     0.534  0.525   0.329
awning-tricycle      220      532        0.42    0.31   0.294   0.199
bus                  131      251        0.866   0.669  0.775   0.601
motor                485      4886       0.72    0.669  0.708   0.379

Speed: 0.6ms preprocess, 18.8ms inference, 0.0ms loss, 1.5ms postprocess per image

```

When making predictions, my code is as follows. I resize any input image to 1280x720. I know that YOLO expects square images by default, but I let Ultralytics handle the padding.

```
from ultralytics import YOLO
model = YOLO("best.pt")
img = cv2.resize(img, (1280, 720), interpolation=cv2.INTER_AREA)
results = model.predict(img, imgsz=1280)
```

The code seems correct, but in some images, it fails to detect objects that should be easily detectable, and sometimes it doesn't detect any objects at all.

When training and performing inference with the default image size of 640 on the same dataset, there are no issues, and object detection generally works fine. The reason I increased the image size to 1280 was to increase the number of detected objects.

Are we sure that the imgsz parameter is working correctly?

### Environment

Ultralytics 8.3.59 🚀 Python-3.9.21 torch-2.0.1+cu118 CUDA:0 (NVIDIA RTX A5000, 24564MiB)
Setup complete ✅ (64 CPUs, 63.7 GB RAM, 4721.2/5588.9 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.9.21
Install             pip
RAM                 63.70 GB
Disk                4721.2/5588.9 GB
CPU                 Intel Xeon Gold 6226R 2.90GHz
CPU count           64
GPU                 NVIDIA RTX A5000, 24564MiB
GPU count           1
CUDA                11.8

### Minimal Reproducible Example

from ultralytics import YOLO

model = YOLO('yolov8x.pt')

model.train(
    data='data/VisDrone.yaml',
    epochs=50,
    imgsz=1280,
    batch=4,
    device=0,
    workers=8,
)

model = YOLO("best.pt")
img = cv2.resize(img, (1280, 720), interpolation=cv2.INTER_AREA)
results = model.predict(img, imgsz=1280)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @uzunkadir, thank you for bringing this up and for your detailed report 🚀! We appreciate your interest in YOLOv8 and your use of the Ultralytics repository. Here are some steps and tips to help us assist you further:

### Documentation and Resources

We recommend checking out our [Docs](https://docs.ultralytics.com/) for specific guidance. You can find examples of how to use the various parameters in both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) formats, which might clarify any potential issues with your configurations. Also, make sure to review our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) to ensure best practices for your experiments.

### Bug Reports 🐛

If this is indeed a bug, we would need a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to debug effectively. Kindly simplify your code to the smallest possible example that replicates the issue. Additionally:
- Ensure your `ultralytics` package is updated to the latest version as shown below to rule out any issues already resolved:
```bash
pip install -U ultralytics
```
- Verify all dependencies are installed correctly, especially [PyTorch>=1.8](https://pytorch.org/get-started/locally/) and CUDA drivers.

### Inference Details 🕵️‍♀️

Given your scenario, where using `imgsz=1280` during inference results in inconsistencies, it would be helpful to:
1. Share specific images where detections are missing. You may either link to sample images or provide anonymized samples illustrative of the issue.
2. Confirm if the padding aligns with expectations by maintaining or altering aspect ratios. This might help isolate any dimension-related issue.
3. Confirm if inference works as expected through the CLI. Use:
```bash
yolo task=detect model=best.pt imgsz=1280 source=path/to/image
```

### Community and Support 🎧

You can access help and share insights with the community in real-time on [Discord](https://discord.com/invite/ultralytics). Alternatively, join in-depth discussions on [Discourse](https://community.ultralytics.com/) or start a conversation with peers on our [Subreddit](https://reddit.com/r/Ultralytics). We're here to support you through all these channels.

### Verified Environments ✅

To confirm the integrity and reproducibility of your task, we recommend experimenting in one of our validated environments:
- **Notebooks** with GPUs included: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud VMs** via GCP or AWS. Refer to our [Google Cloud Quickstart](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker** setups for controlled environments. See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/). <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

---

This is an automated response to ensure you get immediate assistance and direct resources. Rest assured, an Ultralytics engineer will review your issue and respond promptly 😃.

**Comment by Y-T-G** [comment]
It's the wrong resizing method and you shouldn't resize manually.

Use `imgsz=(720, 1280)` instead and pass the original inage.

**Comment by uzunkadir** [comment]
YOLO's default resize method is not working as I expect. Therefore, I apply a better resizing process using `interpolation=cv2.INTER_AREA`, which minimizes the loss of details. Additionally, my images have very different sizes, so I resize them all to the same dimensions. If my images are sized 1280x720, should I specify `imgsz=(720, 1280)` during the prediction process? Should it be the same during training as well? Currently, I have only set `imgsz=1280` during training.

**Comment by Y-T-G** [comment]
You can specify both height and width during prediction and export but not during training
